addons/mrp/wizard/mrp_production_split.py

```diff
@@ -30,12 +32,17 @@ class MrpProductionSplit(models.TransientModel):
         'Split Details', compute="_compute_details", store=True, readonly=False)
     valid_details = fields.Boolean("Valid", compute="_compute_valid_details")
 
+    @api.depends('production_id.lot_producing_id')
+    def _compute_first_lot_name(self):
+        for wizard in self:
+            wizard.first_lot_name = self.env.context.get('default_first_lot_name') or wizard.production_id.lot_producing_id.name
```

Comment:
Is there ever a case when the `default_first_lot_name` wouldn't automatically set the value? And/or do we even need the `default_first_lot_name` , it seems like this value wouldn't be ever be set when `action_prepare_split` is called (i.e. the only place where this default value is set)

```suggestion
        for wizard in self:
            if not wizard.first_lot_name:
                wizard.first_lot_name = wizard.production_id.lot_producing_id.name
```

Reply:
this issue is when `action_generate_first_lot_name` set `first_lot_name`, it won't show on the wizard, I have to return a new action to update this value. This context is added to set the value in the new action